src/scripts/main.js

```diff
@@ -1,3 +1,345 @@
+  select.dataset.qa = inputName;
+
+  offices.forEach((office) => {
+    const option = document.createElement('option');
+
+    option.value = office;
+    option.textContent = office;
+
+    select.appendChild(option);
+  });
+
+  label.appendChild(select);
+  form.appendChild(label);
+}
+
+// function for creating inputs
+function createFormInput(inputName, type) {
+  const label = document.createElement('label');
+  const input = document.createElement('input');
+
+  label.textContent = createFirstBigLetter(inputName) + ':';
+
+  input.type = type;
+  input.name = inputName;
+  input.dataset.qa = inputName;
+
+  label.appendChild(input);
+  form.appendChild(label);
+}
+
+// function for creating labels
+function createFirstBigLetter(input) {
+  return input.charAt(0).toUpperCase() + input.slice(1);
+}
+
+// creating button
+const button = document.createElement('button');
+
+button.type = 'submit';
+button.textContent = 'Save to table';
+
+form.appendChild(button);
+
+// creating notifications
+function createNotification(type, message) {
+  const notification = document.createElement('div');
+  const notificationTitle = document.createElement('h2');
+  const notificationDescription = document.createElement('p');
+
+  notification.dataset.qa = 'notification';
+  notification.classList.add('notification');
+
+  notification.appendChild(notificationTitle);
+  notification.appendChild(notificationDescription);
+
+  notificationDescription.textContent = message;
+
+  notificationTitle.textContent = `${createFirstBigLetter(type)} message:`;
+  notification.classList.add(type);
+
+  document.body.appendChild(notification);
+
+  setTimeout(() => {
+    if (document.body.contains(notification)) {
+      document.body.removeChild(notification);
+    }
+  }, 5000);
+}
+
+// formatting error
+function formatString(input) {
+  const items = input.split(' ');
+
+  if (items.length === 1) {
+    return `${items[0]} is`;
+  }
+
+  if (items.length === 2) {
+    return `${items[0]} and ${items[1]} are`;
+  }
+
+  const allButLast = items.slice(0, -1).join(', ');
+  const last = items[items.length - 1];
+
+  return `${allButLast} and ${last} are`;
+}
+
+// creating errors
+form.addEventListener('submit', (e) => {
+  e.preventDefault();
+
+  const error = document.querySelector('.notification');
+
+  if (error) {
+    document.body.removeChild(error);
+  }
+
+  const formData = new FormData(form);
+
+  let isEmptyField = false;
+  let whichEmptyField = '';
+
+  for (const [key, value] of formData) {
+    if (!value.trim()) {
+      isEmptyField = true;
+      whichEmptyField += `${key} `;
+    }
+  }
+
+  if (isEmptyField) {
+    const emptyFields = formatString(whichEmptyField.trim());
+
+    createNotification(
+      'error',
+      `Failed because ${emptyFields} empty, please enter correct value!`,
+    );
+
+    return;
+  }
+
+  if (Number(formData.get('name').length) < 4) {
+    createNotification('error', 'Name should be greater than 4 letters!');
```

Comment:
This error message might be misleading because the check is for names with fewer than 4 characters, not greater. Consider rephrasing it to 'Name should be at least 4 letters long!'